airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java

```diff
@@ -36,6 +41,8 @@
   private final Integer partSize;
   private final S3FormatConfig formatConfig;
 
+  private final AtomicReference<AmazonS3> s3Client = new AtomicReference<>(null);
```

Comment:
given that we're locking all the get/set operations, why does this need to use an atomic reference at all?